libsolutil/StringUtils.h

```diff
@@ -101,109 +101,122 @@ std::string joinHumanReadablePrefixed
 		return _separator + joinHumanReadable(_list, _separator, _lastSeparator);
 }
 
-/// Same as @ref formatNumberReadable but only for unsigned numbers
+/// Formats large numbers to be easily readable by humans.
+/// Returns decimal representation for smaller numbers; hex for large numbers.
+/// "Special" numbers, powers-of-two and powers-of-two minus 1, are returned in
+/// formulaic form like 0x01 * 2**24 - 1.
+/// @a T can be any integer variable, will typically be u160, u256 or bigint.
+/// @param _value to be formatted
+/// @param _useTruncation if true, internal truncation is also applied,
+/// like  0x5555...{+56 more}...5555
+/// @example formatNumberReadable((u256)0x7ffffff) = "0x08 * 2**24"
+/// @example formatNumberReadable(-57896044618658097711785492504343953926634992332820282019728792003956564819968) = -2**255
 template <class T>
-inline std::string formatUnsignedNumberReadable (
+inline std::string formatNumberReadable(
 	T const& _value,
 	bool _useTruncation = false
 )
 {
 	static_assert(
-		std::is_same<bigint, T>::value || !std::numeric_limits<T>::is_signed,
-		"only unsigned types or bigint supported"
-	); //bigint does not carry sign bit on shift
+		std::numeric_limits<T>::is_integer,
+		"only integer numbers are supported"
+	);
+
+	bool _isNegative = false;
+	bigint v;
+
+	if (_value < 0)
+		_isNegative = true;
 
 	// smaller numbers return as decimal
-	if (_value <= 0x1000000)
+	if (!_isNegative && _value <= 0x1000000)
 		return _value.str();
 
+	if (_isNegative && bigint(-1 * _value) <= 0x1000000)
+		return "-" + (bigint(-1 * _value)).str();
+
 	HexCase hexcase = HexCase::Mixed;
-	HexPrefix prefix = HexPrefix::Add;
+	HexPrefix hexprefix = HexPrefix::Add;
+
+	std::string _sign = _isNegative ? "-" : "";
```

Comment:
Same as before regarding underscore. 
Also, maybe we could declare and define this before so it can be used in the previous `return` values.

Reply:
I defined `sign` after the previous return values because the condition for entering the return statement was to have `isNegative` be true. So there wasn't any need to use the value of sign instead of just adding `"-"` before the value